airflow/models.py

```diff
@@ -3188,7 +3187,7 @@ def following_schedule(self, dttm):
             cron = croniter(self._schedule_interval, dttm)
             following = timezone.make_aware(cron.get_next(datetime), self.timezone)
             return timezone.convert_to_utc(following)
-        elif isinstance(self._schedule_interval, timedelta):
+        elif self._schedule_interval is not None:
```

Comment:
maybe safer as `elif isinstance(self._schedule_interval, (timedelta, relativedelta))`

Reply:
I'd prefer not to hard code the types supported because it undermines Python duck typing and if some other class similar to relativedelta is created this code would need to be updated.  We actually have this use case for a custom fiscal calendar we have implemented.